pandas/core/array_algos/quantile.py

```diff
@@ -1,11 +1,50 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 import numpy as np
 
 from pandas._libs import lib
+from pandas._typing import ArrayLike
 
-from pandas.core.dtypes.common import is_list_like
+from pandas.core.dtypes.common import (
+    is_list_like,
+    is_sparse,
+)
+from pandas.core.dtypes.missing import (
+    isna,
+    na_value_for_dtype,
+)
 
 from pandas.core.nanops import nanpercentile
 
+if TYPE_CHECKING:
+    from pandas.core.arrays import ExtensionArray
+
+
+def quantile_compat(values: ArrayLike, qs, interpolation: str, axis: int) -> ArrayLike:
+    """
+    Compute the quantiles of the given values for each quantile in `qs`.
+
+    Parameters
+    ----------
+    values : np.ndarray or ExtensionArray
+    qs : a scalar or list of the quantiles to be computed
+    interpolation : str
+    axis : int
+
+    Returns
+    -------
+    np.ndarray or ExtensionArray
+    """
+    if isinstance(values, np.ndarray):
+        fill_value = na_value_for_dtype(values.dtype, compat=False)
```

Comment:
would make sense to unify these signature (e.g. put mask / fill_value) as keywords (at the end) or pass to the ea_compat (and ignore them).

i would also compute these *inside* quantile_with_mask (if these are not passed), then thi function becomes way simpler (this can be a followon)

Reply:
worth doing here?

Reply:
im gonna have to do at least one more pass anyway, so id rather get this in and revisit in the next one